### PR TITLE
Print branch-hash only if it was set externally

### PR DESCRIPTION
**Pull request by sergeyfrolov**

### CODE REVIEW

File: tapdance/logger.go [hunk 1]
```diff
@@ -19,14 +19,17 @@ var initLoggerOnce sync.Once
...
 	initLoggerOnce.Do(func() {
 		logrusLogger = logrus.New()
 		logrusLogger.Formatter = new(formatter)
 		logrusLogger.Level = logrus.InfoLevel
-		//logrusLogger.Level = logrus.DebugLevel
-		logrusLogger.Infof("Running gotapdance build %s", build)
+		// logrusLogger.Level = logrus.DebugLevel
+
+		// CI will overwrite the build string with branch and commit hash
+		const build string = "[BUILD]"
```
Comment (ewust) [inline tapdance/logger.go]: Does it actually get overwritten? My cli prints out "Running gotapdance build [BUILD]". Is there a script we're supposed to run to replace this?

Comment (sergeyfrolov) [reply tapdance/logger.go]: Is the comment above the line you just commented ambiguous, or you haven't noticed it?

**Comment by szechuen** [comment]
Instead of suppressing the output when the build hasn't been compiled on CI, could we give Psiphon a separate function that silences/redirects the output either by pre-initializing logrusLogger or setting a variable of sorts? I'm thinking about this in the context of whether we would want to statically include the version of the code eventually, thus propagating it reliably to the output of whichever application is using the library unless they choose to silence it. Even in Psiphon's case, it would be nice to have that information in the log file they are producing. Also open to relocating the print statement somewhere else - it just needs to be somewhere common and early to all interfaces and wrapped in sync.Once.

**Comment by sergeyfrolov** [comment]
Let's try to keep it simple and not force Psiphon to change their code. I don't think we should produce extra functions over things like *this*.

> we would want to statically include the version of the code eventually
> open to relocating the print statement somewhere else

I am also open to relocating it elsewhere (simple solution: use `sync.Once` in Logger's `Format()` func or one of the hooks it provides) whenever we want, but I'd rather do so without perturbing our partners. I'd honestly rather rip this whole feature out instead. `Logger()` function shouldn't be printing anything in the first place, and it was fine for our own debug builds, but we shouldn't be making it someone else's problem now.

**Pull request merged**